fix bad deprecated_api link (#3522)
ROCm-Developer-Tools/HIP: docs/how-to/hip_porting_driver_api.md

HIP provides a context API to facilitate easy porting from existing Driver codes.
In HIP, the Ctx functions largely provide an alternate syntax for changing the active device.

> Most new applications will prefer to use `hipSetDevice` or the stream APIs, therefore HIP has marked `hipCtx` APIs as **deprecated**. Support for these APIs may not be available in future releases. For more details on deprecated APIs please refer [HIP deprecated APIs](../reference/deprecated_api_list).

Reviewer: Seems odd - how can an application "prefer"?
Perhaps this can be reworded.

Author: Reworded

> Most new applications will prefer to use `hipSetDevice` or the stream APIs, therefore HIP has marked `hipCtx` APIs as **deprecated**. Support for these APIs may not be available in future releases. For more details on deprecated APIs please refer [HIP deprecated APIs](../reference/deprecated_api_list).

Reviewer: ```suggestion
Most new applications will prefer to use `hipSetDevice` or the stream APIs, therefore HIP has marked `hipCtx` APIs as **deprecated**. Support for these APIs may not be available in future releases. For more details about deprecated APIs, see [HIP deprecated APIs](../reference/deprecated_api_list).
```
